test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+var bootstrapModeler = require('../../../../helper').bootstrapModeler,
+    camundaModdlePackage = require('camunda-bpmn-moddle/resources/camunda'),
+    coreModule = require('bpmn-js/lib/core').default,
+    inject = require('../../../../helper').inject,
+    modelingModule = require('bpmn-js/lib/features/modeling').default,
+    propertiesPanelModule = require('lib'),
+    propertiesProviderModule = require('lib/provider/camunda'),
+    replaceModule = require('bpmn-js/lib/features/replace').default;
+
+describe('element-templates - replace behavior', function() {
```

Comment:
One thing we could consider before merging it: 

adding a test case that verifies it keeps also applied for a sub-type. E.g. when changing from `bpmn:Task` to `bpmn:ServiceTask`.

Reply:
Good point, I'll take care of it 👍

Reply:
Do you have better test names than those?
```
  element-templates - replace behavior
    ✓ should link to an element of a type the template applies to
    ✓ should link to a sub-element of a type the template applies to
    ✓ should not link to an element of a type the template doesn't apply to
```

Reply:
What about

```
* should keep the template applied
* should keep the template applied - parent type
* should unlink the template if not applicable
```

Reply:
* should not unlink if template can be applied to type
* should not unlink if template can be applied to parent type
* should unlink if template cannot be applied to type